ceph/rados/core_workflows.py

```diff
@@ -1034,18 +1035,206 @@ def get_journalctl_log(
-        - high_recovery_ops
-        - high_client_ops
+    def get_pg_states(self):
+        """
+        Used to get the pg states
+        """
+        cmd = "ceph -s -f json"
+        pgstate_list = []
+        try:
+            output = self.run_ceph_command(cmd=cmd)
+            output_dump = json.dumps(output)
+            output_load_json = json.loads(output_dump)
+            for pg in output_load_json["pgmap"]["pgs_by_state"]:
+                pgstate_list.append(pg["state_name"])
+            return pgstate_list
+        except Exception:
+            log.error("Error while un-setting the flag")
+            return False
 
+    def get_osd_list(self):
+        """
+        Used to get the list of OSDs configured in the cluster nodes
+        Returns:
+            Dictionary with the keys as cluster osd names and values are the list of osd names configured to the node.
+        """
+        try:
+            osd_list = dict()
+            cmd = "ceph osd  tree"
+            osds = self.run_ceph_command(cmd)
+            for entry in osds["nodes"]:
+                if entry["type"] == "host":
+                    osd_list[entry["name"]] = entry["children"]
+            return osd_list
+        except Exception as e:
+            log.error("Failed to  get the osd list : %s", str(e))
+
+    def get_osd_pid(self, host_object, osd):
+        """
+        Used to get the process id of an  OSD
         Args:
-            profile: mclock profile name
-            osd: "osd" service by default or "osd.<Id>"
+             host_object is osd host object
+             osd is the osd number
+        Returns:
+            PID number of the osd
+        """
+        try:
+            get_pid_cmd = (
+                'ps -ef | grep -w "/usr/bin/ceph-osd .* osd.'
+                + osd
+                + '"| grep -v "/run/podman-init" | grep -v grep | awk \'{print $2 }\''
+            )
+            osd_pid = host_object.exec_command(sudo=True, cmd=get_pid_cmd)
+            osd_pid = ("".join(osd_pid)).strip()
+            return osd_pid
+        except Exception as e:
+            log.error("Failed to  get the process id of OSD : %s", str(e))
+
+    def check_osd_status(self, host_object, osd):
         """
-        return self.node.shell([f"ceph config set {osd} osd_mclock_profile {profile}"])
+        Used to check the process is running in osd node or not
+        Args:
+             host_object is osd host object
+             osd is the osd number
+        Returns:
+            If process is up and running then  returns 0 else return non zero
+        """
+        try:
+            osd_status_cmd = (
+                'systemctl is-active --quiet "$(systemctl list-units --type=service | grep -i "ceph-.*@osd.'
+                + osd
+                + ".service\" | awk '{print $1}')\" && echo $?"
+            )
+            status = host_object.exec_command(
+                sudo=True, cmd=osd_status_cmd, long_running=True
+            )
```

Comment:
long_running command gives only return code status.

Reply:
I will implement this while fixing the below.